src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -395,6 +395,13 @@ private void StartWriting(byte[] buffer, int offset, int count, AsyncProtocolReq
 
                 do
                 {
+                    if (count == 0 && !SslStreamPal.CanEncryptEmptyMessage)
```

Comment:
> but I was aiming for "least impactful".

I agree: if the Desktop implementation works with 0-payload-byte TLS records then I don't see why we would remove that behavior. I remember that one of the first attempts at mitigating BEAST was to use 0-payload-byte TLS records. The fix was successful: as @bartonjs mentioned, some apps couldn't interpret them correctly.

@bartonjs I didn't know we supported this: if you have some time, could you please add a test showing the difference in behavior and to lock-in the Windows behavior?